lib/optimizely.rb

```diff
@@ -470,9 +472,14 @@ def get_feature_variable_integer(feature_flag_key, variable_key, user_id, attrib
       variable_value
     end
 
+    def is_valid
```

Comment:
Why not continue using `is_valid` in all the API methods? Seems less repetitive than having all the methods check whether the config itself is valid. We should also check for nil here.